pkg/retagger/job_pattern.go

```diff
@@ -61,25 +61,23 @@ func (job *PatternJob) Compile(r *Retagger) ([]SingleJob, error) {
 	var jobs []SingleJob
 
 	for _, match := range matches {
-		_, exists := existingTagMap[match]
+		j := SingleJob{
+			Source:  job.Source,
+			Options: job.Options,
+		}
+
+		// Override Source options from our pattern.
+		j.Source.Tag = match
+		j.Destination = GetDestinationForJob(&j, r)
```

Comment:
`j.Destination.Tag` will have the suffix at this point if it has a custom image.